### ISSUE TITLE
Implement UI for downloading expression matrices PART 1

### ISSUE THREAD

**Issue by tihuan** [chanzuckerberg/corpora-data-portal#602]
What:
1. Add "Download Dataset" column
2. Add https://blueprintjs.com/docs/#core/components/overlay, so on "Download" button click, it triggers the empty modal. Please see if `Blueprint.js` allows us to import only the `Overlay` component, so we don't add code/components we don't use to the app weight
3. Add `Blueprint.js` to Data Portal, including its css files and other potential requirements: [doc](https://blueprintjs.com/docs/#blueprint/getting-started) -- DONE in feature branch
4. Write e2e tests to assert #1 and #2 work
5. UPDATE: Please do https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/corpora-data-portal/607 as part of this ticket, thank you!

[Part 2 ticket](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/corpora-data-portal/518) will implement the `<DownloadDataset />` component 

=============== ORIGINAL DESCRIPTION BELOW ===============

Designs are now available:

1. Interactive flow: https://chanzuckerberg.invisionapp.com/share/K6YKM66VU8J#/screens/429923777
2. Figma: https://www.figma.com/file/hy3wOqrKbunH1C3lA7CuP6/download-dataset?node-id=0%3A1

Requirements:
1. Implement non-resumable download and CURL command as shown in the mock
2. Add download column in the datasets page (current homepage)
3. Add modal according to the mock
4. Work with BE to figure out the POST endpoint interface `/dataset/{id}` and what the response shape looks like
5. Add E2E tests for the feature
6. Perhaps use `clipboard-copy` for copying CURL command

User flow:
1. When a user visits `cellxgene.cziscience.com/`, they see the datasets with a new column "Download dataset" and its "download" button
2. On "download" button click, a modal shows up with the dataset's information
3. The user can now either copy the CURL command OR click the "download" button to download the dataset

Context:
1. When the user clicks on the "download" button on the dataset row on the homepage, the modal opens and fetches the download link by sending a POST to `/dataset/{id}`, which returns the link as specified in https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/corpora-data-portal/476 (which means we might need loading state)

2. We then use the download link to populate the the CURL command and the download button!

**Comment by tihuan** [comment]
Just updated the description to include https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/corpora-data-portal/607 🙏 Thanks so much for doing this, @Bento007 !!

**Comment by tihuan** [comment]
Since the PR merged into the feature branch instead of main, I kept the status as WIP for now

Once the feature branch is merged, I'll move both #602 and #518 to "ready for staging"! Thank you!
